server/commands/teamUpdater.ts

```diff
@@ -58,10 +61,47 @@ const teamUpdater = async ({ params, user, team, ip }: TeamUpdaterProps) => {
   if (inviteRequired !== undefined) {
     team.inviteRequired = inviteRequired;
   }
+  if (allowedDomains !== undefined) {
```

Comment:
I took a look at the `$get` and `$set` methods but wasn't exactly sure how to use those here to cleanup this code

I think most of the complexity in this code is due to the fact that the frontend sends us an array of strings, and we need to 'translate' those to an array of `TeamDomain` objects.

I could try to use `$add` and `$remove` instead of the creation on line 82 and the deletion on line 98 but didn't seem like that was any simpler when I was playing around with it. Let me know if you disagree and happy to change it!

Or let me know if there is something I am missing where I should be able to use `$set` with an array of domain strings and not `TeamDomain` objects!

Reply:
As it's contained within this command I don't feel too bad about it, you're right there would need to be some translation to model objects either way